dora/core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheFileInStream.java

```diff
@@ -291,8 +291,21 @@ public long remaining() {
 
   @Override
   public int positionedRead(long pos, byte[] b, int off, int len) throws IOException {
-    return readInternal(new ByteArrayTargetBuffer(b, off), off, len, ReadType.READ_INTO_BYTE_ARRAY,
-        pos, true);
+    try {
+      return readInternal(new ByteArrayTargetBuffer(b, off), off, len,
+          ReadType.READ_INTO_BYTE_ARRAY, pos, true);
+    } catch (IOException | RuntimeException e) {
+      LOG.warn("Failed to read from Alluxio's page cache.", e);
```

Comment:
When does this error mostly occur ? If it tends to happen often in certain cases, would the log.warn above be too chatty ? 

Is there any way to instrument (count + expose) the number of such fallbacks. Potentially to highlight that something may not be right ?

Finally, can we catch something smaller than RuntimeException ? Always wary of seeing us silently ignore NPE's for eg :-)

Reply:
There shouldn't be any IO or Runtime exception in the normal read from a local disk.  But I don't have any data so far.  Or we could add a counter and expose it via JMX as you suggested?

I think the major idea is to make presto query won't fail when any IO or runtime exceptions occurs in cache, so I catch all the IO and Runtime exceptions.  That means even though there is a NPE somehow, we should still eat and ignore it, am I right?